pandas/tests/groupby/test_timezone.py

```diff
@@ -0,0 +1,112 @@
+                    None,
+                    Timestamp("2019-01-01 12:00:00"),
+                    Timestamp("2019-01-01 12:30:00"),
+                    None,
+                    None,
+                ]
+            },
+            {
+                "time": [
+                    Timestamp("2019-01-01 12:00:00"),
+                    Timestamp("2019-01-01 12:30:00"),
+                    Timestamp("2019-01-01 14:00:00"),
+                    Timestamp("2019-01-01 14:30:00"),
+                    Timestamp("2019-01-01 14:00:00"),
+                    Timestamp("2019-01-01 14:30:00"),
+                ]
+            },
+            {
+                "time": [
+                    Timestamp("2019-01-01 12:00:00"),
+                    Timestamp("2019-01-01 12:30:00"),
+                    Timestamp("2019-01-01 12:00:00"),
+                    Timestamp("2019-01-01 12:30:00"),
+                    Timestamp("2019-01-01 14:00:00"),
+                    Timestamp("2019-01-01 14:30:00"),
+                ]
+            },
+        ),
+        (
+            {
+                "id": ["A", "B", "A", "B", "A", "B"],
+                "time": [
+                    Timestamp("2019-01-01 12:00:00", tz="Asia/Tokyo"),
+                    Timestamp("2019-01-01 12:30:00", tz="Asia/Tokyo"),
+                    None,
+                    None,
+                    Timestamp("2019-01-01 14:00:00", tz="Asia/Tokyo"),
+                    Timestamp("2019-01-01 14:30:00", tz="Asia/Tokyo"),
+                ],
+            },
+            {
+                "time": [
+                    None,
+                    None,
+                    Timestamp("2019-01-01 12:00:00", tz="Asia/Tokyo"),
+                    Timestamp("2019-01-01 12:30:00", tz="Asia/Tokyo"),
+                    None,
+                    None,
+                ]
+            },
+            {
+                "time": [
+                    Timestamp("2019-01-01 12:00:00", tz="Asia/Tokyo"),
+                    Timestamp("2019-01-01 12:30:00", tz="Asia/Tokyo"),
+                    Timestamp("2019-01-01 14:00:00", tz="Asia/Tokyo"),
+                    Timestamp("2019-01-01 14:30:00", tz="Asia/Tokyo"),
+                    Timestamp("2019-01-01 14:00:00", tz="Asia/Tokyo"),
+                    Timestamp("2019-01-01 14:30:00", tz="Asia/Tokyo"),
+                ]
+            },
+            {
+                "time": [
+                    Timestamp("2019-01-01 12:00:00", tz="Asia/Tokyo"),
+                    Timestamp("2019-01-01 12:30:00", tz="Asia/Tokyo"),
+                    Timestamp("2019-01-01 12:00:00", tz="Asia/Tokyo"),
+                    Timestamp("2019-01-01 12:30:00", tz="Asia/Tokyo"),
+                    Timestamp("2019-01-01 14:00:00", tz="Asia/Tokyo"),
+                    Timestamp("2019-01-01 14:30:00", tz="Asia/Tokyo"),
+                ]
+            },
+        ),
+    ],
+)
+def test_shift_bfill_ffill_tz(data, expected_shift, expected_bfill, expected_ffill):
```

Comment:
Can you put this test in `test_groupby.py`

Also you can parameterize over `shift`, `bfill`, and `ffill`